### PR TITLE
WIP - Fix argument processing in tests

### PR DESCRIPTION
**Pull request by omajid**
[`_.takeRightWhile`](https://lodash.com/docs/#takeRightWhile) stops as soon as the predicate fails. We don't want that. We want to take all options, no matter what.

Please note that no runtime called `dotnet` exists when I run the commands:

# Before:

```
$ yarn ittest --runtimes=dotnet --verbose --timeout 1000000 | tee omajid-test.l og                                                                                                                   
yarn run v1.13.0                                                                                                     
$ NODE_PATH=lib:catalog mocha --opts it-test/mocha.opts -r ts-node/register it-test/all-tests.ts --runtimes=dotnet --v
erbose --timeout 1000000                                                                                             
                                                        
                                                                      
  Backends                                                  
    Runtime nodejs with database,health,rest,welcome       
```

# After:

```
$ yarn ittest --runtimes=dotnet --verbose --timeout 1000000 | tee omajid-test.log                                                                                                                   
yarn run v1.13.0                                                                                                     
$ NODE_PATH=lib:catalog mocha --opts it-test/mocha.opts -r ts-node/register it-test/all-tests.ts --runtimes=dotnet --v
erbose --timeout 1000000                                                                                             
                                                        
                                                                      
  0 passing (2ms)                                                                                                     
                                                                    
Done in 1.70s.                         
```

cc @gastaldi

### CODE REVIEW

**Comment by quintesse** [comment]
Well yes, but this was done for a reason, let me see if I can remember what it was exactly....

**Comment by quintesse** [comment]
Ah yes, remembered now. Thing is that the `yarn ittest` internally gets turned into something like `/usr/bin/node /home/..../node_modules/mocha/bin/_mocha --timeout 10000 --opts it-test/mocha.opts -r ts-node/register it-test/all-tests.ts` so filtering only on `--` also gives us a bunch of options that are not related to our app!

Which is the reason it only looks at the end of the command line.

So my suggestion for this would be:

 - On the CLI pass them as a single flag, eg `--timeout=10000`
 - Don't pass Mocha options on the command line but put them in `it-test/mocha.opts`

Another option would be to rewrite this to `takeRightWhile` until we reach the word `ittest`. A bit hacky but if you really want to be able to write `--timeout 10000` it might work.

File: it-test/functions.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function getServiceName(part: Part) {
 }
 
 function procArgs() {
-    return _.takeRightWhile(process.argv, i => i.startsWith('--'));
```
Comment (quintesse) [inline it-test/functions.ts]: Perhaps change it to `return _.takeRightWhile(process.argv, i => i !== 'ittest');` ?

**Comment by quintesse** [comment]
Btw thanks for catching this! :-)

**Comment by omajid** [comment]
Should I close this PR? This issue doesn't affect me any more. Also, I don't really understand the code here and I expect someone else more familiar can probably do a better set of fixes.

**Comment by quintesse** [comment]
Yes, let's close this. Thanks @omajid !

**Pull request closed without merging**